shared/utility/io/uart.cpp

```diff
@@ -164,10 +162,13 @@ namespace utility::io {
         return !(fcntl(fd, F_GETFL) < 0 && errno == EBADF);
     }
 
-    ssize_t uart::read(void* buf, size_t count) {
+    ssize_t uart::read(void* buf, size_t count) const {
```

Comment:
I would say this shouldn't be const either because reading will change the state of this object (next read will return different data)